core/jobs/batch_jobs/topic_migration_jobs.py

```diff
@@ -0,0 +1,307 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs used for migrating the topic models."""
+
+from __future__ import annotations
+
+import logging
+
+from core import feconf
+from core.domain import caching_services
+from core.domain import topic_domain
+from core.domain import topic_fetchers
+from core.domain import topic_services
+from core.jobs import base_jobs
+from core.jobs.io import ndb_io
+from core.jobs.transforms import job_result_transforms
+from core.jobs.types import job_run_result
+from core.platform import models
+
+import apache_beam as beam
+import result
+
+from typing import Iterable, Sequence, Tuple
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import base_models
+    from mypy_imports import datastore_services
+    from mypy_imports import topic_models
+
+(base_models, topic_models) = models.Registry.import_models([
+    models.Names.BASE_MODEL, models.Names.TOPIC])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class MigrateTopicJob(base_jobs.JobBase):
```

Comment:
Please modify the job so that it returns both the counts of migrated and previously migrated activities (you can look into other jobs). Also, we should modify the job in a way that when `_migrate_topic` returns even one error then we should not migrate anything.

Reply:
@vojtechjelinek  I have added the counts for migrated and previously migrated activities. Regarding not migrating anything when `_migrate_topic` returns an error, I was thinking we could check the count for `Err` in the results `PCollection` and maybe quit the job if the count is valid.

Reply:
Makes sense